### ISSUE TITLE
Add header ignore option to readability-duplicate-include

### ISSUE THREAD

**Issue by nick-potenski** [llvm/llvm-project#166938]
Our codebase has a few cases where duplicate includes are expected.  For example, we have a pair of headers to manage enabling and disabling structure packing which are typically included around a structure that we want packed on a byte boundary.

```c
#include "pack_begin.h"
typedef struct {
} a_packed_struct;
#include "pack_end.h"

typedef struct {
} an_unpacked_struct;

#include "pack_begin.h"  //  readability-duplicate-include warns here
typedef struct {
} another_packed_struct;
#include "pack_end.h" //  readability-duplicate-include warns here
``` 

It would be nice if this checker provided a configuration option taking a list of strings or regexes defining headers that are OK to duplicate.

**Comment by nick-potenski** [comment]
I was really hoping for a better solution than NOLINT considering how many instances of this exist in our codebase.  NOLINT seems like it should be used for the exception, not an established pattern.

**Comment by lakshsidhu04** [comment]
I would like to work on this one!

**Comment by firewave** [comment]
> I don't see how you're going to avoid the false positives. How can you detect this pattern reliably? It would seem to require pretty sophisticated analysis of the included file.

Wouldn't it "just" be some of the logic from `llvm-header-guard` but inverted?

**Comment by lakshsidhu04** [comment]
Guys I am sorry. I no longer plan to work on this issue. Could someone please unassign me?

**Comment by Priyanshu3820** [comment]
I would like to work on this issue. @EugeneZelenko please assign me.
